packages/build-tools/src/utils/cacheKey.ts

```diff
@@ -18,6 +18,11 @@ export const CACHE_KEY_PREFIX_BY_PLATFORM: Record<Platform, string> = {
   [Platform.IOS]: IOS_CACHE_KEY_PREFIX,
 };
 
+export const PUBLIC_CACHE_KEY_PREFIX_BY_PLATFORM: Record<Platform, string> = {
+  [Platform.ANDROID]: `public-${ANDROID_CACHE_KEY_PREFIX}`,
+  [Platform.IOS]: `public-${IOS_CACHE_KEY_PREFIX}`,
```

Comment:
This feels tiny bit weird, like changing `IOS_CACHE_KEY_PREFIX` is going to change both of those keys? It feels a bit accidental to me.

Also, do we use `IOS_CACHE_KEY_PREFIX` anywhere? Let's not export both `IOS_CACHE_KEY_PREFIX` and `CACHE_KEY_PREFIX_BY_PLATFORM[Platform.IOS]`? Maybe we can only use the latter and remove the former?